planetiler-core/src/test/java/com/onthegomap/planetiler/stream/WriteableJsonStreamArchiveTest.java

```diff
@@ -0,0 +1,225 @@
+package com.onthegomap.planetiler.stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.collect.ImmutableMap;
+import com.onthegomap.planetiler.archive.TileArchiveMetadata;
+import com.onthegomap.planetiler.archive.TileCompression;
+import com.onthegomap.planetiler.archive.TileEncodingResult;
+import com.onthegomap.planetiler.config.Arguments;
+import com.onthegomap.planetiler.geo.TileCoord;
+import com.onthegomap.planetiler.util.LayerStats;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.locationtech.jts.geom.CoordinateXY;
+import org.locationtech.jts.geom.Envelope;
+
+class WriteableJsonStreamArchiveTest {
+
+  private static final StreamArchiveConfig defaultConfig = new StreamArchiveConfig(false, Arguments.of());
+  private static final TileArchiveMetadata maxMetadataIn =
+    new TileArchiveMetadata("name", "description", "attribution", "version", "type", "format", new Envelope(0, 1, 2, 3),
+      new CoordinateXY(1.3, 3.7), 1.0, 2, 3,
+      List.of(
+        new LayerStats.VectorLayer("vl0",
+          ImmutableMap.of("1", LayerStats.FieldType.BOOLEAN, "2", LayerStats.FieldType.NUMBER, "3",
+            LayerStats.FieldType.STRING),
+          Optional.of("description"), OptionalInt.of(1), OptionalInt.of(2)),
+        new LayerStats.VectorLayer("vl1",
+          Map.of(),
+          Optional.empty(), OptionalInt.empty(), OptionalInt.empty())
+      ),
+      ImmutableMap.of("a", "b", "c", "d"),
+      TileCompression.GZIP);
+  private static final String maxMetadataOut = """
```

Comment:
Can we always expect the metadata to be serialized in this exact order? I usually do something like parse both to an object and ensure the objects are equal just in case key ordering changes in the future.

Reply:
ok got it, thanks! Good to know about ImmutableMap preserving order.